api/src/controllers/users.js

```diff
@@ -92,21 +133,43 @@ async function loginUser(req, res) {
  * @sends: Updates the user object
  */
 async function updateUser(req, res) {
-  const { id: _id } = req.params;
-  const user = req.body;
-
-  if (!mongoose.Types.ObjectId.isValid(_id))
-    return res.status(404).send("Error: No user with that id.");
-
-  try {
-    const updatedUser = await User.findByIdAndUpdate(_id, user, {
-      new: true,
-      runValidators: true,
-    });
-    res.status(200).json(updatedUser);
-  } catch (err) {
-    res.status(400).json(`Error: ${err.message}`);
-  }
+	const { id: _id } = req.params;
+	const user = req.body;
+
+	if (!mongoose.Types.ObjectId.isValid(_id))
+		return res.status(404).send("Error: No user with that id.");
+
+	try {
+		const updatedUser = await User.findByIdAndUpdate(_id, user, {
+		new: true,
+		runValidators: true,
+		});
+		res.status(200).json(updatedUser);
+	} catch (err) {
+		res.status(400).json(`Error: ${err.message}`);
+	}
+}
+
+/**
+ * Handles a GET request to get the currently logged in user's details on the endpoint /users/current.
+ *
+ * @param req request object
+ * @param res response object
+ * @sends: The user object of the currently logged in user
+ */
+async function getCurrentUser(req, res) {
+	if (req.user) {
+		const user = {
+		"_id": req.user["_id"],
+		givenName: req.user.givenName,
+		familyName: req.user.familyName,
+		email: req.user.email
+		}
+
+		res.json(user);
+	} else {
+		res.status(400).json({message: "User not logged in"});
+	}
```

Comment:
We could probably remove this function entirely, since once the user is logged in, we store their details in the redux global store.

Reply:
@kdav108 the main point of this function was to act as a demonstration of how the authentication works.
Eg, if a user is not logged in, this function would return an error.
So I think its worth keeping as an example, so people can see how to use the authFunctions.ensureAuthenticated thing.

Reply:
Ahhh I see, that makes sense. We'll keep it then